flash/core/data/io/input.py

```diff
@@ -40,6 +39,13 @@
     IterableDataset = object
 
 
+def _deepcopy_dict(nested_dict: Any) -> Any:
+    """Utility to deepcopy a nested dict."""
+    if not isinstance(nested_dict, Dict):
+        return nested_dict
```

Comment:
Nice! Just curious, and shouldn't block this PR getting merged, did you happen to do any benchmarking, or even an idea on how much improvement we get?

Reply:
Nope, no idea haha, just a hunch 😃

Reply:
Intuitions work 🔥